gap/oper.gi

```diff
@@ -608,15 +608,19 @@ end);
 InstallMethod(OnDigraphs, "for a mutable digraph by out-neighbours and a perm",
 [IsMutableDigraph and IsDigraphByOutNeighboursRep, IsPerm],
 function(D, p)
-  local out;
-  if ForAll(DigraphVertices(D), i -> i ^ p = i) then
+  local out, permed;
+  if p = () then
     return D;
-  elif ForAny(DigraphVertices(D), i -> i ^ p > DigraphNrVertices(D)) then
+  fi;
+
+  out := D!.OutNeighbours;
+  permed := Permuted(out, p);
+  if Length(permed) > DigraphNrVertices(D) then
```

Comment:
If `p`  is invalid, `Permuted` simply won't work properly (I think) and cause its own `Error` to happen, which on the one hand is good because it means that the method prevents something improper happen, but it does mean that we lose the ability to give a proper error message.